# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserParent.h
# Commit: bce3f99441c0
# Full Hash: bce3f99441c0dccc5171cc8df20b1d86313dc478
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2022-08-03 03:42:48
# Regressor Bug: 1781129
# File Overlap Count: 1
# Description:
#   Bug 1781129 - Part 3: Remove Shmem overload of GetSurfaceData, r=edgar
#   
#   After the previous changes there was only one consumer left of the Shmem
#   version of GetSurfaceData, which could easily be changed to use BigBuffer,
#   removing the need for that overload.
# ==============================================================================

diff -r fc135243503e -r bce3f99441c0 dom/ipc/BrowserParent.h
--- a/dom/ipc/BrowserParent.h	Tue Aug 02 17:15:41 2022 +0000
+++ b/dom/ipc/BrowserParent.h	Tue Aug 02 17:15:42 2022 +0000
@@ -675,7 +675,7 @@
 
   mozilla::ipc::IPCResult RecvInvokeDragSession(
       nsTArray<IPCDataTransfer>&& aTransfers, const uint32_t& aAction,
-      Maybe<Shmem>&& aVisualDnDData, const uint32_t& aStride,
+      Maybe<BigBuffer>&& aVisualDnDData, const uint32_t& aStride,
       const gfx::SurfaceFormat& aFormat, const LayoutDeviceIntRect& aDragRect,
       nsIPrincipal* aPrincipal, nsIContentSecurityPolicy* aCsp,
       const CookieJarSettingsArgs& aCookieJarSettingsArgs,